client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -1926,13 +1924,58 @@ public void shutdown() {
   }
 
   @Override
-  public void setupLifecycleManagerRef(String host, int port) {
+  public void setupLifecycleManagerRef(String host, int port) throws CelebornIOException {
     logger.info("setupLifecycleManagerRef: host = {}, port = {}", host, port);
-    lifecycleManagerRef =
-        rpcEnv.setupEndpointRef(new RpcAddress(host, port), RpcNameConstants.LIFECYCLE_MANAGER_EP);
+    try {
+      lifecycleManagerRef =
+          callLifecycleManagerWithTimeoutRetry(
+              () ->
+                  rpcEnv.setupEndpointRef(
+                      new RpcAddress(host, port), RpcNameConstants.LIFECYCLE_MANAGER_EP),
+              "setupLifecycleManagerRef");
+    } catch (Exception e) {
+      logger.error("setupLifecycleManagerRef failed, host = {}, port = {}", host, port);
+      throw new CelebornIOException("setupLifecycleManagerRef failed", e);
+    }
     initDataClientFactoryIfNeeded();
   }
 
+  public <T> T callLifecycleManagerWithTimeoutRetry(Callable<T> callable, String name)
```

Comment:
Instead of making changes everywhere - do we want to simply change askSync/askAsync to become retry aware ? With number of retries passed in as a param (for specific cases where we dont want retries for ex) ?

Reply:
I agree to change askSync/askAsync.